sopel/lifecycle.py

```diff
@@ -0,0 +1,180 @@
+"""Deprecation module for Sopel developers and Plugin authors.
+
+.. versionadded:: 8.0
+
+    Previously in :mod:`sopel.tools`, the :func:`deprecated` function has been
+    moved to this newly created module, as it can be used in every part of the
+    Sopel codebase, including :mod:`sopel.tools` itself.
+
+"""
+from __future__ import annotations
+
+import functools
+import inspect
+import logging
+import traceback
+from typing import Callable, Optional
+
+from pkg_resources import parse_version
```

Comment:
This made me realize we don't actually have an explicit dependency on `pkg_resources` in either requirements file, and we should—_especially_ if it's used at runtime like this.

While it's _probably_ going to be available if people have `setuptools` available to install Sopel in the first place, we shouldn't implicitly rely on anything. Zen of Python, etc.

Reply:
Yep, to be replaced by `importlib.metadata` per my personal tickler list.